cmd/goal/clerk.go

```diff
@@ -151,13 +151,20 @@ var clerkCmd = &cobra.Command{
 	},
 }
 
-func waitForCommit(client libgoal.Client, txid string) error {
+func waitForCommit(client libgoal.Client, txid string, transactionLastValidRound uint64) error {
 	// Get current round information
 	stat, err := client.Status()
 	if err != nil {
 		return fmt.Errorf(errorRequestFail, err)
 	}
 
+	// check if we've already committed to the block number equals to the transaction's last valid round.
+	// if this is the case, the transaction would not be included in the blockchain, and we can exit right
+	// here.
+	if stat.LastRound >= transactionLastValidRound {
```

Comment:
Why this check here? 
If the txn is accepted, but the node is past transactionLastValidRound, it will correctly return. 
But, wouldn't the error errorTransactionExpired be wrong?

Reply:
yes, I think that you're correct. If the transaction was sent, we have to test it's acceptance using the `client.PendingTransactionInformation` before testing for expiration.